devices/pine64-pinephone/README.adoc

```diff
@@ -27,7 +71,7 @@ NixOS boot partition.
 
 ```
  $ nix-build --argstr device pine64-pinephone -A build.boot-partition
- $ dd if=result/mobile-nixos-boot.img of=/dev/mmcblkXp3 bs=8M oflag=sync,direct status=progress
+ $ dd if=result/mobile-nixos-boot.img of=/dev/mmcblkXpY bs=8M oflag=sync,direct status=progress
 ```
```

Comment:
It would probably be best to document how to do this with full disk encryption. Users running FDE following this instruction will end up with a non-bootable system.

Not sure if this is the best way, but I did this with:
```
NIX_PATH=nixos-config=/path/to/configuration.nix:nixpkgs=/path/to/nixpkgs:mobile-nixos=/path/to/mobile-nixos \
nix-build --attr config.mobile.outputs.u-boot.boot-partition \<nixpkgs/nixos\>█ 
```
(pulled from installer)

Reply:
Right, these instructions are annoying... I kinda wanted to remove them outright.

The problem is handling of the boot image is still an open question, and I don't know what I want to do with it.

Reply:
I guess what could help here is titling the section *development*, or something like that. Introduce the section with a blurb about how the result here may not apply for the installed system.

Also noting that it's not only for FDE that it will break. The label for the ext4 filesystem will not match either!

Reply:
I'd argue we should have a valid example for installer-based systems as `nixos-rebuild` does not currently update the kernel (at least with my PPP). Creating a new boot-partition image is the next-best way to update the kernel, and thus worth having a supported process for.

Clarifying the incompatibility would at least prevent people from breaking their stage-1s though.

Reply:
Oh, the thought wasn't in the message, but yes, it should be documented how to "reach" the boot image properly from the running system.

Reply:
Your comments are still right, but note that the default setup works with "stage-0", which in turn means the boot image is not required to be updated, since the system will `kexec` into the generation's kernel.

So this is less required than expected.

Hopefully we can tie this loose-end by the time updating the image is required. (And/or move to UEFI-based boot with proper display init.)

Reply:
> the default setup works with "stage-0", which in turn means the boot image is not required to be updated, since the system will `kexec` into the generation's kernel.

This had me somewhat confused at first as my system was clearly not doing this. Looks like this is just because the pinephone pro isn't currently set up this way, and setting `mobile.quirks.supportsStage-0 = true` results in a kernel panic on boot.

Makes sense. Sounds like I should be figuring out how to get stage-0 style boot working with pinephone pro.